tools/src/main/java/org/apache/kafka/trogdor/workload/ProduceBenchWorker.java

```diff
@@ -179,18 +182,34 @@ protected synchronized void delay(long amount) throws InterruptedException {
 
         private final PayloadIterator values;
 
+        private final TransactionGenerator txActionGenerator;
+
         private final Throttle throttle;
 
+        private Iterator<TopicPartition> partitionsIterator;
+        private Future<RecordMetadata> sendFuture;
+        private AtomicInteger transactionsCommitted;
+        private boolean toUseTransactions = false;
+
         SendRecords(HashSet<TopicPartition> activePartitions) {
             this.activePartitions = activePartitions;
+            this.partitionsIterator = activePartitions.iterator();
             this.histogram = new Histogram(5000);
+
+            this.txActionGenerator = spec.transactionGenerator();
+            this.transactionsCommitted = new AtomicInteger();
+            if (txActionGenerator.action() == TransactionAction.INIT_TRANSACTIONS)
+                toUseTransactions = true;
```

Comment:
I think we should check to make sure the first action is either `INIT_TRANSACTIONS` or `NO_OP`.  Other values should not be allowed.